bemani/backend/iidx/copula.py

```diff
@@ -288,306 +296,378 @@ def game_to_db_rank(self, game_dan: int, cltype: int) -> int:
         else:
             raise Exception('Invalid cltype!')
 
-    def handle_IIDX23shop_request(self, request: Node) -> Optional[Node]:
-        method = request.attribute('method')
+    def game_to_db_chart(self, db_chart: int) -> int:
+        return {
+            self.GAME_CHART_TYPE_B7: self.CHART_TYPE_B7,
+            self.GAME_CHART_TYPE_N7: self.CHART_TYPE_N7,
+            self.GAME_CHART_TYPE_H7: self.CHART_TYPE_H7,
+            self.GAME_CHART_TYPE_A7: self.CHART_TYPE_A7,
+            self.GAME_CHART_TYPE_N14: self.CHART_TYPE_N14,
+            self.GAME_CHART_TYPE_H14: self.CHART_TYPE_H14,
+            self.GAME_CHART_TYPE_A14: self.CHART_TYPE_A14,
+        }[db_chart]
+
+    def handle_IIDX23shop_getname_request(self, request: Node) -> Optional[Node]:
+        root = Node.void('IIDX23shop')
+        root.set_attribute('cls_opt', '0')
+        machine = self.data.local.machine.get_machine(self.config.machine.pcbid)
+        root.set_attribute('opname', machine.name)
+        root.set_attribute('pid', str(self.get_machine_region()))
+        return root
 
-        if method == 'getname':
-            root = Node.void('IIDX23shop')
-            root.set_attribute('cls_opt', '0')
-            machine = self.data.local.machine.get_machine(self.config.machine.pcbid)
-            root.set_attribute('opname', machine.name)
-            root.set_attribute('pid', str(self.get_machine_region()))
-            return root
+    def handle_IIDX23shop_savename_request(self, request: Node) -> Optional[Node]:
```

Comment:
Technically all of these can just be a `Node` return instead of an `Optional[Node]` return now that we split the handlers up. I don't think it should hold up the review tho unless you want to fix. I can go back and fix it myself if not.

Reply:
Oh thanks for catching this, I legitimately just forgot...

Reply:
NBD, but tighter typing always helps. In this case since not adding a return implicitly returns `None`, tightening the typing to `Node` instead of `Optional[Node]` will catch if you make a mistake and forget a return either in the default path or in some obscure early-return.